nesta/production/luigihacks/elasticsearchplus.py

```diff
@@ -19,15 +23,144 @@
 COUNTRY_TAG="terms_of_countryTags"
 PUNCTUATION = re.compile(r'[a-zA-Z\d\s:]').sub('', string.printable)
 
+def sentence_chunks(text, chunksize=2000, delim='. '):
+    """Split a string into chunks, but breaking only on 
+    the specified delimiter.
+    
+    Args:
+        text (str): A string to split into chunks
+        chunksize (int): Minimum chunk size to yield
+        delim (str): Delimiter to split chunks
+    Yields:
+        chunk (str): The smallest possible chunks (minimum sise 
+                     :pyobject:`chunksize`) of text.
+    """
+    chunks = [] 
+    for _text in text.split(delim): 
+        chunks.append(_text) 
+        if sum(len(c) for c in chunks) > chunksize: 
+            yield delim.join(chunks) 
+            chunks = [] 
+    if len(chunks) > 0: 
+        yield delim.join(chunks) 
+
+
+class MLStripper(HTMLParser):
+    """Taken from https://stackoverflow.com/questions/753052/strip-html-from-strings-in-python. Tested in _sanitize_html."""
+    def __init__(self):
+        self.reset()
+        self.strict = False
+        self.convert_charrefs= True
+        self.fed = []
+        super().__init__()
+    def handle_data(self, d):
+        self.fed.append(d)
+    def get_data(self):
+        return ''.join(self.fed)
+
+def strip_tags(html):
+    """Taken from https://stackoverflow.com/questions/753052/strip-html-from-strings-in-python. Tested in _sanitize_html"""
+    s = MLStripper()
+    s.feed(html)
+    return s.get_data()
+
+@retry(wait_random_min=20, wait_random_max=30, 
+       stop_max_attempt_number=10)
+def translate(text, translator, chunksize=2000, rate_limit=0.4):
```

Comment:
you didn't use your ratelimit decorator :( 

https://github.com/nestauk/nesta/blob/master/nesta/packages/decorators/ratelimit.py

Reply:
Fine - I put it in :)